subsys/bluetooth/host/hci_core.c

```diff
@@ -765,6 +765,24 @@ static void update_pending_id(struct bt_keys *keys, void *data)
 }
 #endif
 
+static struct bt_conn *find_pending_connect(bt_addr_le_t *peer_addr)
+{
+	struct bt_conn *conn;
+
+	/*
+	 * Make lookup to check if there's a connection object in
+	 * CONNECT or DIR_ADV state associated with passed peer LE address.
+	 */
+	conn = bt_conn_lookup_state_le(peer_addr, BT_CONN_CONNECT);
+	if (conn) {
+		return conn;
+	}
+
+	conn = bt_conn_lookup_state_le(peer_addr, BT_CONN_CONNECT_DIR_ADV);
+
+	return conn;
```

Comment:
Minor: this could just be `return bt_conn_lookup_state_le(peer_addr, BT_CONN_CONNECT_DIR_ADV);`